src/error_measures.jl

```diff
@@ -0,0 +1,165 @@
+
+export ErrorMeasures
+
+"""
+$(TYPEDSIGNATURES)
+Create a struct containing the error measures detailed in Williamson et al. 1992, J Computational Physics
+`reference` is a function of time, longitude and latitude (both in degrees)
+$(TYPEDFIELDS)"""
+@kwdef mutable struct ErrorMeasures <: SpeedyWeather.AbstractCallback
+    reference::Function
```

Comment:
in Julia you can think of every function as being a type of its own. In that sense `reference::Function` is an abstract field -- which you can use, no doubt, but the Julian way of doing this would be to make the type parametric on that function, i.e.

```julia
struct ErrorMeasures{F} <: ...
    reference::F
    ...
```

that way the compiler knows about the function already from the type of `ErrorMeasures` and can compile it away (e.g. inline, etc)